spring-integration-core/src/main/java/org/springframework/integration/config/xml/DefaultInboundChannelAdapterParser.java

```diff
@@ -76,6 +79,7 @@ protected BeanMetadataElement parseSource(Element element, ParserContext parserC
 								+ "element " + IntegrationNamespaceUtils.createElementDescription(element) + ".", source);
 				return null;
 			}
+			Assert.state(innerBeanDef != null, "No inner bean definition found"); //Required because JSpecify does not know how to use handle a preparsed boolean
```

Comment:
I think `Objects.requireNonNull()` would read better.
We definitely never going to have it as null when we reach this place.